configure.ac

```diff
@@ -1978,7 +1978,7 @@ then
   fi
 fi
 LLVM_PROF_ERR=no
-case $CC in
+case $( basename "$CC" ) in
```

Comment:
GNU Autoconf [discourages the use of `basename`](https://www.gnu.org/software/autoconf/manual/autoconf-2.71/html_node/Limitations-of-Usual-Tools.html#index-basename):

> Not all hosts have a working basename. You can use expr instead.

opengroup.org suggests one of the following:

```console
$ which gcc
/usr/bin/gcc
$ a=$(which gcc)
$ expr $a : '.*/\(.*\)' \| $a  # <= either this option
gcc
$ expr "//$a" : '.*/\(.*\)'    # <= or this
gcc
```

They suggest a preference for the latter:

> [...] is a better representation of the previous example. The addition of the "//" characters eliminates any ambiguity about the division operator and simplifies the whole expression. Also note that pathnames may contain characters contained in the IFS variable and should be quoted to avoid having "$a" expand into multiple arguments.

Now, I'm not sure how relevant the GNU Autoconf warning is anymore; I would guess that most systems have a working `basename`. _But_, if it can save us from a future bug report ... :)

How about this:

```suggestion

# GNU Autoconf recommends the use of expr instead of basename.
AS_VAR_SET([CC_BASENAME], [expr "//$CC" : '.*/\(.*\)'])
case "$CC_BASENAME" in
```

... and then of course refer to `CC_BASENAME` in the following switches.

Reply:
Applied in b1f129c1b45a74d044da8f1442215c3a58f5e095